akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -745,15 +745,15 @@ private[stream] object MaterializerSession {
 /**
  * INTERNAL API
  */
-private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Module, val initialAttributes: Attributes) {
+abstract class MaterializerSession(val topLevel: StreamLayout.Module, val initialAttributes: Attributes) {
   import StreamLayout._
 
   // the contained maps store either Subscriber[Any] or VirtualPublisher, but the type system cannot express that
-  private var subscribersStack: List[ju.Map[InPort, AnyRef]] =
+  var subscribersStack: List[ju.Map[InPort, AnyRef]] =
```

Comment:
@drewhk my preference was protected as well and I agree it lays bare internal state.  StreamLayoutSpec - checkMaterialized and FlatTestMaterialized drove some of the exposure after my MaterializerSession changes. Let me take another pass at this.